Framework/Kernel/inc/MantidKernel/cow_ptr.h

```diff
@@ -64,6 +64,8 @@ template <typename DataType> class cow_ptr {
   std::mutex copyMutex;
 
 public:
+  cow_ptr(ptr_type &&resourceSptr);
+  cow_ptr(DataType *resourcePtr);
```

Comment:
Comparing this with http://en.cppreference.com/w/cpp/memory/shared_ptr/shared_ptr (I guess it makes sense to follow this closely):

- `cow_ptr(DataType *resourcePtr);` should be `explicit`
- it would be convenient to also have `constexpr shared_ptr( std::nullptr_t );`. Would that break anything? The behavior of the default constructor would still be different, but at least we would have to option to create a `cow_ptr` that does not manage anything.